File: app/payload_methods_frontend/Post.java
// Step 1: Define Post Model
data class Post(val imageURL: String, val spotifyId: String, val title: String, val userId: Int, val timestamp: Long)

// Step 2: Implement Post Generation Logic
fun generatePost(imageURL: String, spotifyId: String, title: String, userId: Int): Post {
        val timestamp = System.currentTimeMillis()
        return Post(imageURL, spotifyId, title, userId, timestamp)
        }

// Step 3 & 4: Integration with UI and Display Generated Posts (for console application)
        fun main() {
        // Simulate user input or generate data
        val imageURL = "https://example.com/image.jpg"
        val spotifyId = "spotify:track:4uLU6hMCjMI75M1A2tKUQC"
        val title = "My Awesome Playlist"
        val userId = 123 // Example user ID

        // Generate a post
        val generatedPost = generatePost(imageURL, spotifyId, title, userId)

        // Display the generated post
        println("New Post:")
        println("Image URL: ${generatedPost.imageURL}")
        println("Spotify ID: ${generatedPost.spotifyId}")
        println("Title: ${generatedPost.title}")
        println("User ID: ${generatedPost.userId}")
        println("Timestamp: ${generatedPost.timestamp}")
        }

File: app/Post.kt
import java.util.Calendar

// Step 1: Define Models
data class Post(val imageURL: String, val spotifyId: String, val title: String, val userId: Int, val timestamp: Calendar)

data class User(val displayName: String, val spotifyId: String)

data class Comment(val comment: String, val postId: Int, val userId: Int)
// Step 2: Implement Post Generation Logic
fun createPost(imageURL: String, spotifyId: String, title: String, userId: Int): Post {
    val timestamp = Calendar.getInstance()
    return Post(imageURL, spotifyId, title, userId, timestamp)
}

fun newUser(displayName: String, spotifyId: String): User{
    return User(displayName, spotifyId)
}

fun createComment(comment: String, postId: Int, userId: Int): Comment{
    return Comment(comment, postId, userId)
}
// Step 3 & 4: Integration with UI and Display Generated Posts
fun main() {
}
